network/transport/grpc/tls_offloading.go

```diff
@@ -78,13 +80,39 @@ func (t *tlsOffloadingAuthenticator) authenticate(serverStream grpc.ServerStream
 	if err != nil {
 		return nil, errors.New("TLS client header escaping is invalid")
 	}
-	certificates, err := core.ParseCertificates([]byte(unescaped))
+
+	var certificates []*x509.Certificate
+	if strings.Contains(unescaped, "-----BEGIN CERTIFICATE-----") {
+		certificates, err = t.parsePEMCert(unescaped)
+	} else {
+		certificates, err = t.parseDERCert(values[0])
+	}
 	if err != nil {
-		return nil, fmt.Errorf("invalid client certificate(s) in header: %w", err)
+		return nil, err
 	}
 	if len(certificates) != 1 {
 		return nil, fmt.Errorf("expected exactly 1 client certificate in header, found %d", len(certificates))
 	}
+	return certificates, err
+}
+
+func (t *tlsOffloadingAuthenticator) parseDERCert(data string) ([]*x509.Certificate, error) {
+	bytes, err := base64.StdEncoding.DecodeString(data)
+	if err != nil {
+		return nil, fmt.Errorf("unable to base64 decode client cert header: %w", err)
+	}
```

Comment:
the (DER + )base64 encoding comes from the `[ssl_c_der,base64]` in the HAProxy config:

```
backend grpc_servers
         mode http
         http-request set-header X-SSL-CERT %{+Q}[ssl_c_der,base64]
         server node1 nuts_node:5555 check proto h2
```

It is possible to specify other encodings in haproxy.
not sure if this should be more generic or that we can just specify that we expect DER+base64

Reply:
That's right, I see you could also use hex encoding. I don't think we need to be more generic; base64 is probably the most common occurrence. I'll add it to the documentation.